src/client/src/rt-components/platform/adapters/openfin/excel/index.ts

```diff
@@ -0,0 +1,119 @@
+import { formTable } from './utils'
+import { ExcelWorkbook, ExcelWorksheet, ExcelInterface } from './types'
+import { InteropTopics } from 'rt-components'
+
+/*
+  ACTIONS
+  init           - initializes the Excel object - called on application connect/through rt-launcher
+  openExcel      - opens the Excel, exits Menu view then opens the template
+  publishToExcel - cells set here + polling incase of Excel open after/before launch
+
+  UTILS
+  addOnCloseListener - cleans up, sets Excel variables to null on Excel close
+  setWorkVariables   - the only location where Excel variables are set
+  isAlreadyRunning   - 1. checks if Excel object initialized
+                       2. checks if Excel connected (opened)
+                       3. returns Excel workbook object if name matches EXCEL_SHEET_NAME
+  pollForExcelOpen   - called repeatedly if no workbook exists - if the correct workbook
+                       is found the Excel variables are set            
+*/
+
+const EXCEL_HOST_URL = 'http://adaptiveconsulting.github.io/ReactiveTraderCloud/excel'
+const EXCEL_SHEET_NAME = 'RTExcel.xlsx'
+
+enum ExcelPageSetup {
+  Blotter = 'A2',
+  Poisitins = 'A2',
+  CcyPairs = 'G2',
+}
+
+class ExcelAdapter implements ExcelInterface {
```

Comment:
Just for convention sake, you should move this class from the index file and into its own file. And then import and export it from here.

Reply:
moved out of index now